core/src/main/java/de/jplag/options/JPlagOptions.java

```diff
@@ -49,15 +49,16 @@
  * @param mergingOptions are the options related to the subsequence match merging mechanism that opposed obfuscation.
  * @param normalize enables additional normalization mechanisms. Only supported by some language modules.
  * @param analyzeComments If true, comments will be extracted from the submissions.
+ * @param skipVersionCheck If true, skips checking GitHub API for the latest version.
  */
 @RecordBuilder()
 public record JPlagOptions(@JsonSerialize(using = LanguageSerializer.class) Language language, Integer minimumTokenMatch,
         @JsonSerialize(contentUsing = FileSerializer.class) Set<File> submissionDirectories,
         @JsonSerialize(contentUsing = FileSerializer.class) Set<File> oldSubmissionDirectories,
         @JsonSerialize(using = FileSerializer.class) File baseCodeSubmissionDirectory, String subdirectoryName, List<String> fileSuffixes,
         String exclusionFileName, SimilarityMetric similarityMetric, double similarityThreshold, int maximumNumberOfComparisons,
-        ClusteringOptions clusteringOptions, boolean debugParser, MergingOptions mergingOptions, boolean normalize, boolean analyzeComments)
-        implements JPlagOptionsBuilder.With {
+        ClusteringOptions clusteringOptions, boolean debugParser, MergingOptions mergingOptions, boolean normalize, boolean analyzeComments,
+        boolean skipVersionCheck) implements JPlagOptionsBuilder.With {
```

Comment:
I'm unsure whether I would add this to the JPlag options , I'd argue in the parser it would be enough as it does not affect the core but only the cli.

@tsaglam @robinmaisch what do you think ?

Reply:
I have not had time to look at the PR, but if the version check is executed in the CLI only (as I remember it to be), then it does not make sense in the options.